hygieia-jenkins-plugin/src/main/java/hygieia/builder/ArtifactBuilder.java

```diff
@@ -66,30 +89,44 @@ private void buildArtifacts() {
                 bac.setArtifactVersion(version);
                 bac.setCanonicalName(f.getName());
                 bac.setArtifactName(HygieiaUtils.getFileNameMinusVersion(f, version));
-                bac.setTimestamp(build.getTimeInMillis());
-                bac.setBuildId(buildId);
-                CommitBuilder commitBuilder = new CommitBuilder(build);
+                bac.setTimestamp(run.getTimeInMillis());
+                bac.setBuildId(hygieiaBuildId);
+
+                if (run instanceof WorkflowRun) {
+                    changeLogSets = ((WorkflowRun) run).getChangeSets();
+                } else if (run instanceof AbstractBuild) {
+                    changeLogSets = ((AbstractBuild) run).getChangeSets();
+                }
+                CommitBuilder commitBuilder = new CommitBuilder(changeLogSets);
                 bac.getSourceChangeSet().addAll(commitBuilder.getCommits());
-                
-                bac.getMetadata().put("buildUrl", HygieiaUtils.getBuildUrl(build));
-                bac.getMetadata().put("buildNumber", HygieiaUtils.getBuildNumber(build));
-                bac.getMetadata().put("jobUrl", HygieiaUtils.getJobUrl(build));
-                bac.getMetadata().put("jobName", HygieiaUtils.getJobName(build));
-                bac.getMetadata().put("instanceUrl", HygieiaUtils.getInstanceUrl(build, listener));
-                
-                String scmUrl = HygieiaUtils.getScmUrl(build, listener);
-                String scmBranch = HygieiaUtils.getScmBranch(build, listener);
-                String scmRevisionNumber = HygieiaUtils.getScmRevisionNumber(build, listener);
-                
-                if (scmUrl != null) { bac.getMetadata().put("scmUrl", scmUrl); }
-                if (scmBranch != null) { 
-                	if (scmBranch.startsWith("origin/")) {
-                		scmBranch = scmBranch.substring(7);
-                	}
-                	bac.getMetadata().put("scmBranch", scmBranch); 
+
+                bac.getMetadata().put("buildUrl", HygieiaUtils.getBuildUrl(run));
+                bac.getMetadata().put("buildNumber", HygieiaUtils.getBuildNumber(run));
+                bac.getMetadata().put("jobUrl", HygieiaUtils.getJobUrl(run));
+                bac.getMetadata().put("jobName", HygieiaUtils.getJobName(run));
+                bac.getMetadata().put("instanceUrl", HygieiaUtils.getInstanceUrl(run, listener));
+
+
+                if (run instanceof AbstractBuild) {
+                    AbstractBuild abstractBuild = (AbstractBuild) run;
+                    String scmUrl = HygieiaUtils.getScmUrl(abstractBuild, listener);
```

Comment:
Right now we don't with the normal jenkins job so you can just use the first one listed. If you feel adventurous though just add a number to the end of each one after the first one : https://github.com/capitalone/Hygieia/issues/1048#issuecomment-268658328